app/containers/wallet/dialogs/WalletTrezorConnectDialogContainer.js

```diff
@@ -0,0 +1,81 @@
+// @flow
+import React, { Component } from 'react';
+import { observer, inject } from 'mobx-react';
+
+import environment from '../../../environment';
+import type { InjectedDialogContainerProps } from '../../../types/injectedPropsType';
+
+import AboutDialog from '../../../components/wallet/trezorConnect/AboutDialog';
+import ConnectDialog from '../../../components/wallet/trezorConnect/ConnectDialog';
+import SaveDialog from '../../../components/wallet/trezorConnect/SaveDialog';
+
+import { ProgressStepOption } from '../../../stores/ada/TrezorConnectStore';
+
+type Props = InjectedDialogContainerProps;
+@inject('stores', 'actions') @observer
+export default class WalletTrezorConnectDialogContainer extends Component<Props> {
+
+  static defaultProps = { actions: null, stores: null, children: null, onClose: () => {} };
+
+  cancel = () => {
+    this.props.onClose();
+    this._getAction().cancel.trigger();
+  };
+
+  render() {
+    const store = this._getStore();
+    const action = this._getAction();
+
+    let component = null;
+
+    switch (store.progressInfo.currentStep) {
+      case ProgressStepOption.ABOUT:
+        component = (
+          <AboutDialog
+            progressInfo={store.progressInfo}
+            isActionProcessing={store.isActionProcessing}
+            error={store.error}
+            submit={action.submitAbout.trigger}
+            cancel={this.cancel}
+          />);
+        break;
+      case ProgressStepOption.CONNECT:
+        component = (
+          <ConnectDialog
+            progressInfo={store.progressInfo}
+            isActionProcessing={store.isActionProcessing}
+            error={store.error}
+            goBack={action.goBacktToAbout.trigger}
+            submit={action.submitConnect.trigger}
+            cancel={this.cancel}
+          />);
+        break;
+      case ProgressStepOption.SAVE:
+        component = (
+          <SaveDialog
+            progressInfo={store.progressInfo}
+            isActionProcessing={store.isActionProcessing}
+            error={store.error}
+            defaultWalletName={store.defaultWalletName}
+            submit={action.submitSave.trigger}
+            cancel={this.cancel}
+          />);
+        break;
+      default:
+        console.error('Error : something unexpected happened');
+        break;
+    }
+
+    return component;
+  }
+
+  /** Returns the store which is responsible for this Container */
+  _getStore() {
+    return this.props.stores.substores[environment.API].trezorConnect;
+  }
+
+  /** Returns the action which is responsible for this Container */
+  _getAction() {
```

Comment:
missing return type

Reply:
I was thinking that, we should have one to one relationship between Container and Action + Store in longer term.

If we use one to one relationship then things will be more clear about responsibility and relationship between Container, Action and Store.

Anyway, for now i will make it more specific.

Reply:
DONE